CMakeLists.txt

```diff
@@ -198,124 +193,197 @@ else()
+    PRIVATE
+        ${PTLS_INCLUDE_DIRS}
+        ${OPENSSL_INCLUDE_DIR}
+    PUBLIC
+        picoquic)
 
 target_link_libraries(picoquic-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-)
+    PRIVATE
+        ${OPENSSL_LIBRARIES}
+    PUBLIC
+        ${PTLS_LIBRARIES}
+        Threads::Threads)
+
+set_picoquic_build_options(picoquic-core)
 
-add_library(picoquic-log
-    ${LOGLIB_LIBRARY_FILES}
-)
+add_library(picoquic-log ${LOGLIB_LIBRARY_FILES})
 
-target_include_directories(picoquic-log PUBLIC loglib)
+target_include_directories(picoquic-log
+    PRIVATE
+        ${PTLS_INCLUDE_DIRS}
+    PUBLIC
+        picoquic
+        loglib)
 
-add_library(picohttp-core
-    ${PICOHTTP_LIBRARY_FILES}
-)
+add_library(picohttp-core ${PICOHTTP_LIBRARY_FILES})
 
 target_link_libraries(picohttp-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-)
+    PRIVATE
+        ${PTLS_LIBRARIES}
+        ${OPENSSL_LIBRARIES}
+    PUBLIC
+        picoquic-core)
+
+target_include_directories(picohttp-core
+    PRIVATE
+        ${PTLS_INCLUDE_DIRS}
+        ${OPENSSL_INCLUDE_DIR}
+    PUBLIC
+        picoquic)
+
+set_picoquic_build_options(picohttp-core)
 
 add_executable(picoquicdemo
     picoquicfirst/picoquicdemo.c
-    picoquicfirst/getopt.c
-)
+    picoquicfirst/getopt.c)
 
 target_link_libraries(picoquicdemo
-    picoquic-log
-    picoquic-core
-    picohttp-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
-
-target_include_directories(picoquicdemo PRIVATE loglib picohttp)
-
-add_executable(picolog_t
-    picolog/picolog.c
-)
+    PRIVATE
+        ${PTLS_LIBRARIES}
+        ${OPENSSL_LIBRARIES}
+        #${CMAKE_DL_LIBS}
+        picoquic-log
+        picoquic-core
+        picohttp-core)
+
+target_include_directories(picoquicdemo
+    PRIVATE
+        #${PTLS_INCLUDE_DIRS}
+        #${OPENSSL_INCLUDE_DIR}
+        #picoquic
+        #loglib
+        picohttp)
+
+set_picoquic_build_options(picoquicdemo)
+
+add_executable(picolog_t picolog/picolog.c)
 
 target_link_libraries(picolog_t
-    picoquic-log
-    picoquic-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
-
-target_include_directories(picolog_t PRIVATE loglib)
+    PRIVATE
+        #${PTLS_LIBRARIES}
+        #${OPENSSL_LIBRARIES}
+        #${CMAKE_DL_LIBS}
+        picoquic-log
+        picoquic-core)
 
-add_executable(picoquic_ct picoquic_t/picoquic_t.c
-    ${PICOQUIC_TEST_LIBRARY_FILES}
-)
+target_include_directories(picolog_t
+    PRIVATE
+        #picoquictest
+        loglib)
 
-target_link_libraries(picoquic_ct
-    picoquic-log
-    picoquic-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
+set_picoquic_build_options(picolog_t)
 
-target_include_directories(picoquic_ct PRIVATE loglib)
+add_executable(picoquic_ct
+    picoquic_t/picoquic_t.c
+    ${PICOQUIC_TEST_LIBRARY_FILES})
 
```

Comment:
I know it is written like that in the current CMakeList, but this leads to compiling the test libraries twice, for `picoquic_ct` and for `picohttp_ct`.